### PR TITLE
feat: Quality tasks and quality action

### PR DESCRIPTION
**Pull request by johanseto**
## Description

The PR purpose is to integrate quality tasks into the repo.

## Additional information
This PR is based on a fork made in my profile. 
The initial  conversation is inthe PR: https://github.com/eduNEXT/codejailservice/pull/3
## Checklist for Merge

- [x] Update makefile
- [x] Add lint or quality action.
- [x] Clean python code
- [ ] Rebased master/main
- [ ] Squashed commits

### CODE REVIEW

**Comment by johanseto** [comment]
I test the changes in my local environment at this point and codejailservice works. The local container was configured with help of @ericfab179 using this service. 
Also, I remove `codejailservice/tutor.py` because my container doesn't present any error or change.

Next: 
- Test in other environments.
- Setup PR and test PR.

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,53 @@
...
+	pydocstyle codejailservice *.py
+	isort --check-only --diff --recursive codejailservice *.py
+	make selfcheck
+
+requirements: ## install development environment requirements
+	pip install -r requirements/pip.txt
+	pip install -r requirements/pip-tools.txt
+	pip-sync requirements/dev.txt
+
+
```
Comment (felipemontoya) [inline Makefile:49]: Remove this empty lines

File: Makefile [hunk 2]
```diff
@@ -1,7 +1,53 @@
...
+	find codejailservice -name '__pycache__' -exec rm -rf {} +
+	find codejailservice -name '*.pyc' -exec rm -f {} +
+	find codejailservice -name '*.pyo' -exec rm -f {} +
+	find codejailservice -name '*~' -exec rm -f {} +
+	coverage erase
+	rm -fr build/
+	rm -fr dist/
+	rm -fr *.egg-info
+
+
```
Comment (felipemontoya) [inline Makefile:26]: extra empty line

File: Makefile [hunk 3]
```diff
@@ -1,7 +1,53 @@
+.PHONY: quality  help requirements\
+	piptools selfcheck  upgrade test clean
```
Comment (felipemontoya) [inline Makefile]: typo: two spaces

File: codejailservice/app.py [hunk 4]
```diff
@@ -56,22 +54,41 @@ def configure_codejail(app):
                 limit_overrides_context=context,
             )
 
+
+# -----------------app configuration-------------------------
```
Comment (felipemontoya) [inline codejailservice/app.py]: This long kind of messages have a special syntax. See:  https://github.com/edx/edx-platform/blob/master/lms/envs/common.py#L57

File: codejailservice/app.py [hunk 5]
```diff
@@ -56,22 +54,41 @@ def configure_codejail(app):
...
+
+app = Flask(__name__)
```
Comment (felipemontoya) [inline codejailservice/app.py:58]: I'm not soo familiar with flask, but this general settings seem weird to me in the same app file. Can we move them to a different file that does the config and start?

File: codejailservice/app.py [hunk 6]
```diff
@@ -56,22 +54,41 @@ def configure_codejail(app):
...
+                       "codejailservice.config.ProductionConfig")
+app.config.from_object(env_config)
+
 configure_codejail(app)
 
-from codejail.safe_exec import SafeExecException, json_safe
-from codejail.safe_exec import not_safe_exec as codejail_not_safe_exec
-from codejail.safe_exec import safe_exec as codejail_safe_exec
+# Lazy imports of codejail functions according(after) the configuration  app.
+code_jail_safe_exec = importlib.import_module('codejail.safe_exec')
```
Comment (felipemontoya) [inline codejailservice/app.py]: This lazy import looks funky as well, maybe by moving to the other file we can make it cleaner

File: requirements/dev.in [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+-r base.txt
+
+pylint
+pycodestyle
+pydocstyle
+coverage
+pytest
+pudb
+
+
```
Comment (felipemontoya) [inline requirements/dev.in]: extra line

**Review by felipemontoya — CHANGES_REQUESTED** [review]
Left inline comments.
We still need to find a better way to configure the app without the lazy import

File: codejailservice/app.py (hunk 4 above)
Comment (johanseto) [reply codejailservice/app.py]: Yes, I changed taking care the django syntax `######### FEAT #############`. Also I have a codestyle issue E266 for too many `#`. 
image

File: requirements/dev.in (hunk 7 above)
Comment (johanseto) [reply requirements/dev.in]: Solved in 2a4a46f.

File: Makefile (hunk 3 above)
Comment (johanseto) [reply Makefile]: Solved in 2a4a46f

File: Makefile (hunk 2 above)
Comment (johanseto) [reply Makefile:26]: Solved in 2a4a46f

File: Makefile (hunk 1 above)
Comment (johanseto) [reply Makefile:49]: Solved in 2a4a46f

File: codejailservice/app.py (hunk 4 above)
Comment (johanseto) [reply codejailservice/app.py]: In this commit 2a4a46f ,  I can pass the pycodestyle given an extra whitespace. 
`# ################ FEAT ############` . What's your opinion about it?

File: codejailservice/app.py (hunk 5 above)
Comment (johanseto) [reply codejailservice/app.py:58]: In this commit e4c6724, I propose a change to move routes to their folder. This uses "blueprint" of flask.
In that case, this file(`app.py`) only configures the app. The run file is `wsgi.py`

File: codejailservice/app.py (hunk 6 above)
Comment (johanseto) [reply codejailservice/app.py]: For this, I make a method inside the `routes/code-exec.py` call `import_code_jail_safe_exec`.  e4c6724

**Comment by johanseto** [comment]
I test the proposal for routes distribution in my local environment and codejailservice works. e4c6724
Next:

- Test in other environments.
- Setup PR and test PR.

**Comment by JuanDavidBuitrago** [comment]
These tasks were solved in:

https://github.com/eduNEXT/codejailservice/pull/10
https://github.com/eduNEXT/codejailservice/pull/11
https://github.com/eduNEXT/codejailservice/pull/12

**Pull request closed without merging**